lesson_6/edmond_sabou/functions2.py

```diff
@@ -0,0 +1,123 @@
+    ('MT', ['75', '77', '78', '80', '81', '81', '85', '84', '86']),
+    ('NL', ['94', '94', '95', '96', '96', '97', '98', '98', '98']),
+    ('NO', ['92', '93', '94', '93', '97', '97', '97', '96', '98']),
+    ('PL', ['67', '70', '72', '75', '76', '80', '82', '84', '87']),
+    ('PT', ['58', '61', '62', '65', '70', '74', '77', '79', '81']),
+    ('RO', ['47', '54', '58', '61', '68', '72', '76', '81', '84']),
+    ('RS', ['0', '0', '0', '0', '64', '0', '68', '73', '80']),
+    ('SE', ['91', '92', '93', '90', '91', '94', '95', '93', '96']),
+    ('SI', ['73', '74', '76', '77', '78', '78', '82', '87', '89']),
+    ('SK', ['71', '75', '78', '78', '79', '81', '81', '81', '82']),
+    ('TR', ['0', '47', '49', '60', '70', '76', '81', '84', '88']),
+    ('UK', ['83', '87', '88', '90', '91', '93', '94', '95', '96']),
+    ('XK', ['0', '0', '0', '0', '0', '0', '89', '93', '93']),
+]
+
+
+#######################################################################################################################
+def create():
+    new_dict = {country: data_list for country, data_list in raw_data}
+
+    return {
+        country: [
+            {"year": "2011", "coverage": data_list[0]}, {"year": "2012", "coverage": data_list[1]},
+            {"year": "2013", "coverage": data_list[2]}, {"year": "2014", "coverage": data_list[3]},
+            {"year": "2015", "coverage": data_list[4]}, {"year": "2016", "coverage": data_list[5]},
+            {"year": "2017", "coverage": data_list[6]}, {"year": "2018", "coverage": data_list[7]},
+            {"year": "2019", "coverage": data_list[8]}
+        ]
+        for country, data_list in new_dict.items()
+    }
+
+
+#######################################################################################################################
+#
+# get_year_data(data_set, "2019")
+#
+# >>> {'2019': [('Romania', 84), ('Germany', 95), ..., ('Latvia', 85)]}
+#
+def get_year_data(new_data_set, year):
+    #    print(type(new_data_set))
+    #    print(new_data_set)
+    new = {
+        year: [(country, index["coverage"])
+               for country in new_data_set.keys()
+               for index in new_data_set[country]
+               if index["year"] == str(year)]  # Q: De ce este nevoie sa pun str(year)??
+    }  # fara str imi returneaza asta -> {2019: []}
+
+    return new
+
+
+#######################################################################################################################
+# get_country_data(dataset, "Romania")
+#
+# >>> {'Romania': [('2019', 84), ('2018', 86), ..., ('2011', 72)]}
+#
+def get_country_data(new_data_set, country):
+    new2 = {
+        country: [(value["year"], value["coverage"])
+                  for value in new_data_set[country]]
+    }
+
+    return new2
+
+
+#######################################################################################################################
+def perform_average(country_data):  # e urat.. stiu, non-pythonic, dar numai asa am reusit

```

Comment:
the logic is good, try to leave the comments in English 😄  and it is preferred to have the comment above the function definition

Reply:
understood